binders/kafka-binder/spring-cloud-stream-binder-kafka-streams/src/test/java/org/springframework/cloud/stream/binder/kafka/streams/InteractiveQueryServiceMultiStateStoreTests.java

```diff
@@ -0,0 +1,235 @@
+				.web(WebApplicationType.NONE)
+				.run("--server.port=0",
+						"--spring.jmx.enabled=false",
+						"--spring.cloud.stream.function.definition=app1;app2",
+						"--spring.cloud.stream.function.bindings.app1-in-0=input1",
+						"--spring.cloud.stream.function.bindings.app2-in-0=input2",
+						"--spring.cloud.stream.kafka.streams.binder.functions.app1.application-id=stateStoreTestApp1",
+						"--spring.cloud.stream.kafka.streams.binder.functions.app2.application-id=stateStoreTestApp2",
+						"--spring.cloud.stream.kafka.streams.binder.configuration.application.server="
+								+ embeddedKafka.getBrokersAsString(),
+						"--spring.cloud.stream.kafka.streams.binder.brokers="
+								+ embeddedKafka.getBrokersAsString())
+		) {
+			waitForRunningStreams(context.getBean(KafkaStreamsRegistry.class));
+			// The KafkaStreams.store() used by query service is non-deterministic so perform the operation multiple times to
+			// surface any possible issues. Also, no need to actually write anything to the stores, the store.get() call will
+			// cause a failure when the state store is invalid.
+			InteractiveQueryService queryService = context.getBean(InteractiveQueryService.class);
+			for (int i = 0; i < 100; i++) {
+				assertThat(queryService.getQueryableStore(STORE_1_NAME, QueryableStoreTypes.keyValueStore())
+						.get("someKey")).isNull();
+				assertThat(queryService.getQueryableStore(STORE_2_NAME, QueryableStoreTypes.keyValueStore())
+						.get("someKey")).isNull();
+			}
+		}
+	}
+
+	@Test
+	void stateStoreNotAvailableThrowsException() {
+		try (ConfigurableApplicationContext context = new SpringApplicationBuilder()
+				.sources(MultipleAppsWithUnusedStateStoresTestApplication.class)
+				.web(WebApplicationType.NONE)
+				.run("--server.port=0",
+						"--spring.jmx.enabled=false",
+						"--spring.cloud.stream.function.definition=app1;app2",
+						"--spring.cloud.stream.function.bindings.app1-in-0=input1",
+						"--spring.cloud.stream.function.bindings.app2-in-0=input2",
+						"--spring.cloud.stream.kafka.streams.binder.functions.app1.application-id=stateStoreTestApp3",
+						"--spring.cloud.stream.kafka.streams.binder.functions.app2.application-id=stateStoreTestApp4",
+						"--spring.cloud.stream.kafka.streams.binder.configuration.application.server="
+								+ embeddedKafka.getBrokersAsString(),
+						"--spring.cloud.stream.kafka.streams.binder.brokers="
+								+ embeddedKafka.getBrokersAsString())
+		) {
+			waitForRunningStreams(context.getBean(KafkaStreamsRegistry.class));
+
+			InteractiveQueryService queryService = context.getBean(InteractiveQueryService.class);
+
+			assertThatThrownBy(() -> queryService.getQueryableStore(STORE_1_NAME, QueryableStoreTypes.keyValueStore()))
+					.isInstanceOf(IllegalStateException.class)
+					.hasMessage("Error retrieving state store: " + STORE_1_NAME)
+					.hasRootCauseInstanceOf(UnknownStateStoreException.class);
```

Comment:
You will notice that I backed off of the specific message verification as was done in main. This is because our message is different than the message spit out by KafkaStreams 3.0. Same exception type, slightly different message. 
```
SCSt 3.2.x + KafkaStreams 3.0 -> old message
SCSt3.2.x + KafkaStreams 3.1+ -> new message
```
I think this is fine as an exception message should not be considered public API.